File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/WeatherRepositoryInterface.java
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository;

import androidx.lifecycle.LiveData;

import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.WeatherResponse;
import retrofit2.Response;

public interface WeatherRepositoryInterface {

    LiveData<Response<WeatherResponse>> getWeather(double lat, double lon);

    void insertWeather(double lat, double lon, double temp, String location);
}

File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/WeatherRepositoryInterface.kt
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository

import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.WeatherResponse
import retrofit2.Response


interface WeatherRepositoryInterface {
    suspend fun getWeather(lat: Double, lon: Double): Response<WeatherResponse>
    suspend fun insertWeather(lat: Double, lon: Double, temp: Double, location: String)
}